### PR TITLE
Fix examples not loading LoRA adapter weights from checkpoint

### LINKED ISSUE #12689: LoRA adapter weights are not loaded from checkpoints
### Describe the bug

examples/text_to_image/train_text_to_image_lora.py file has a serious bug.
The training pipeline saves the accelerator states and the lora adapters correctly while saving checkpoint.   
But the checkpoint loading code **skips loading lora adapters**.

I am raising a PR very soon...
Please look into it.

### Reproduction

Saving checkpoint script (correct):

if global_step % args.checkpointing_steps == 0:
                    if accelerator.is_main_process:
                        # _before_ saving state, check if this save would set us over the `checkpoints_total_limit`
                        if args.checkpoints_total_limit is not None:
                            checkpoints = os.listdir(args.output_dir)
                            checkpoints = [d for d in checkpoints if d.startswith("checkpoint")]
                            checkpoints = sorted(checkpoints, key=lambda x: int(x.split("-")[1]))

                            # before we save the new checkpoint, we need to have at _most_ `checkpoints_total_limit - 1` checkpoints
                            if len(checkpoints) >= args.checkpoints_total_limit:
                                num_to_remove = len(checkpoints) - args.checkpoints_total_limit + 1
                                removing_checkpoints = checkpoints[0:num_to_remove]

                                logger.info(
                                    f"{len(checkpoints)} checkpoints already exist, removing {len(removing_checkpoints)} checkpoints"
                                )
                                logger.info(f"removing checkpoints: {', '.join(removing_checkpoints)}")

                                for removing_checkpoint in removing_checkpoints:
                                    removing_checkpoint = os.path.join(args.output_dir, removing_checkpoint)
                                    shutil.rmtree(removing_checkpoint)

                        save_path = os.path.join(args.output_dir, f"checkpoint-{global_step}")
                        accelerator.save_state(save_path)

                        unwrapped_unet = unwrap_model(unet)
                        unet_lora_state_dict = convert_state_dict_to_diffusers(
                            get_peft_model_state_dict(unwrapped_unet)
                        )

                        StableDiffusionPipeline.save_lora_weights(
                            save_directory=save_path,
                            unet_lora_layers=unet_lora_state_dict,
                            safe_serialization=True,
                        )

                        logger.info(f"Saved state to {save_path}")

Loading script(buggy):
if args.resume_from_checkpoint:
        if args.resume_from_checkpoint != "latest":
            path = os.path.basename(args.resume_from_checkpoint)
        else:
            # Get the most recent checkpoint
            dirs = os.listdir(args.output_dir)
            dirs = [d for d in dirs if d.startswith("checkpoint")]
            dirs = sorted(dirs, key=lambda x: int(x.split("-")[1]))
            path = dirs[-1] if len(dirs) > 0 else None

        if path is None:
            accelerator.print(
                f"Checkpoint '{args.resume_from_checkpoint}' does not exist. Starting a new training run."
            )
            args.resume_from_checkpoint = None
            initial_global_step = 0
        else:
            accelerator.print(f"Resuming from checkpoint {path}")
            accelerator.load_state(os.path.join(args.output_dir, path))
            global_step = int(path.split("-")[1])

            initial_global_step = global_step
            first_epoch = global_step // num_update_steps_per_epoch
    else:
        initial_global_step = 0

### Logs

```shell

```

### System Info

🤗 Diffusers version: 0.35.2
- Platform: Linux-6.6.105+-x86_64-with-glibc2.35
- Running on Google Colab?: Yes
- Python version: 3.12.12
- PyTorch version (GPU?): 2.8.0+cu126 (True)
- Flax version (CPU?/GPU?/TPU?): 0.10.7 (gpu)
- Jax version: 0.7.2
- JaxLib version: 0.7.2
- Huggingface_hub version: 0.36.0
- Transformers version: 4.57.1
- Accelerate version: 1.11.0
- PEFT version: 0.17.1
- Bitsandbytes version: not installed
- Safetensors version: 0.6.2
- xFormers version: not installed
- Accelerator: Tesla T4, 15360 MiB
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

_No response_

### PR DESCRIPTION
**Pull request by SurAyush**
# What does this PR do?

This PR fixes an issue where LoRA adapter weights are **not loaded** when resuming training from a checkpoint in the training example `train_text_to_image_lora.py`.

Currently, when using `--resume_from_checkpoint`, the script restores the UNet state, optimizer, and scheduler states — but **does not restore `pytorch_lora_weights.safetensors`**, even though the checkpoint folder contains it.

This results in resumed training *not applying the previously trained LoRA weights*, causing quality degradation and training inconsistency.

This PR adds the missing logic to:
- detect `pytorch_lora_weights.safetensors` in the checkpoint folder  
- load it using `safetensors.torch.load_file()`  
- apply it to the UNet via `unwrap_model(unet).load_state_dict(..., strict=False)`  
- warn if LoRA weights are missing  
- log success when loaded  

This makes checkpoint resume behavior consistent and correct.

---

# Fixes

Fixes #12689 

---

# Before submitting

- [x] This PR doesn't only fix docs or typos.
- [x] I read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md).
- [x] I checked that only the intended file is modified.
- [x] The change was discussed or motivated by an issue.
- [x] The example was tested with and without `--resume_from_checkpoint`.

---

# Who can review?

- @sayakpaul (Training examples)
- @yiyixuxu (General functionality)

### CODE REVIEW

**Comment by SurAyush** [comment]
Hi! Tagging @sayakpaul and @yiyixuxu for review since this PR modifies a training example.  
Thanks!

**Comment by SurAyush** [comment]
Great idea to update the accelerate hooks to save model. 
working on it sir!

**Comment by SurAyush** [comment]
@sayakpaul 
Added changes as per requirement. 
Also, manually tested them with a small load and save example. Works fine!
Please check.

**Comment by SurAyush** [comment]
Hello!
Tagging @sayakpaul for review.

File: examples/text_to_image/train_text_to_image_lora.py [hunk 1]
```diff
@@ -708,6 +721,56 @@ def collate_fn(examples):
...
+            unexpected_keys = getattr(incompatible_keys, "unexpected_keys", None)
+            # throw warning if some unexpected keys are found and continue loading
+            if unexpected_keys:
+                logger.warning(
+                    f"Loading adapter weights from state_dict led to unexpected keys not found in the model: "
+                    f" {unexpected_keys}. "
+                )
+
+        # Make sure the trainable params are in float32
+        if args.mixed_precision in ["fp16", "bf16"]:
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_lora.py]: We don't need to upcast when using Bfloat16.

File: examples/text_to_image/train_text_to_image_lora.py [hunk 2]
```diff
@@ -37,20 +37,33 @@
...
 from diffusers.utils.import_utils import is_xformers_available
 from diffusers.utils.torch_utils import is_compiled_module
 
+# compatible with both older and new version of diffusers -- more robust approach
+try:
+    from diffusers.loaders.lora_conversion_utils import \
+        convert_unet_state_dict_to_peft
+except ImportError:
+    try:
+        from diffusers.loaders.peft import convert_unet_state_dict_to_peft
```
Comment (sayakpaul) [inline examples/text_to_image/train_text_to_image_lora.py]: It's available from `diffusers.utils`.

**Comment by SurAyush** [comment]
Hello @sayakpaul !!!
I think it is alright now...

**Review by sayakpaul — APPROVED** [review]
Thanks!
